### DISCUSSION TITLE
[Feature] improve reverse geocoding: use OSM, search nearest street, import existing reverse geocoding tags

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#6137] [Feature Request]
### The feature

Especially in sparsely populated areas with large distances between towns, when places have been incorporated and become districts, the results of the reverse geocoding are often not so good. Places are decoded as some other places at a distance of some kilometers. Maybe the used method is looking for the nearest center of a city, but this result could be wrong. A search for the nearest address / street could be better, providing the right administrative information.

Especially in this case, different county are part of the same city.

I compare immich reverse geocoding with reverse geocoding, provided with digiKam, using OSM (OpenStreetMap). These results are correct. But it can take a long time. It is also possible to set the language.

Maybe immich could also use OSM to provide better results? Because it is a one time process, it should be OK to take a lot of time. And this could be an option

It would be also fine, if immich could use existing reverse geocoding tags. For example, digiKam uses a template to store this metadata, the user could configure the template structure in immich and immich could search for and use it.

for example
![image](https://github.com/immich-app/immich/assets/9020187/cc2a717c-e291-4400-9b45-12d076f07fba)

here are some examples of wrong decoding:
(The issue that "Wittstock/Dosse" is interpreted as a nested tag is a digiKam issue)

53.2863, 12.3514
immich: Bütow
![image](https://github.com/immich-app/immich/assets/9020187/5cd9b6f1-69f3-4941-9dc3-ceef84aabde7)

53.2863, 12.3538
immich: Wredenhagen
![image](https://github.com/immich-app/immich/assets/9020187/c7a9b177-a1a5-4dac-89dd-5fd6bea3619b)

53.2866, 12.3543
immich: Bütow
![image](https://github.com/immich-app/immich/assets/9020187/4199c973-f706-4d89-bf9c-69c9a6312afb)

53.2517, 12.6562
immich: Lärch
![image](https://github.com/immich-app/immich/assets/9020187/0d6c4022-7ac9-4f31-b478-36ec5e687183)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by zackpollard** [comment]
The reverse geocoding API using OSM data is only available through Nomatim which we would likely flood with requests if we were to start using their service. We moved away from OSM's own map hosting service for a similar reason, we no longer thought it was fair to be using their service for a project of this size. We already have plans to further improve reverse geocoding, allowing the user to turn on more detailed reverse geocoding that will be a bit slower and also be bigger on disk, but nobody is actively working on that right now although it is on my (ever increasing) list of things to do 😄 I'm going to close this for now as I don't think this is the way we will go with reverse geocoding.

**Comment by MichelZ** [comment]
I've built exactly this now, it uses overture maps and uses a place's geometry/bounding box, not just a point on the map to calculate the place: 
Docs: https://immich-reversegeo.github.io/
GitHub: https://github.com/immich-reversegeo/immich-reversegeo
